src/app/domain/authentication/services/SynchronizeService.ts

```diff
@@ -1,14 +1,45 @@
 import Minilog from '@cozy/minilog'
+import { getDeviceName } from 'react-native-device-info'
 
 import type CozyClient from 'cozy-client'
+import { getErrorMessage } from 'cozy-intent'
 
-const log = Minilog('SynchronizeService')
+import { SOFTWARE_NAME } from '/libs/constants'
+
+export const syncLog = Minilog('📱 SynchronizeService')
 
 export const synchronizeDevice = async (client: CozyClient): Promise<void> => {
   try {
     await client.getStackClient().fetchJSON('POST', '/settings/synchronized')
-    log.info('📱 Device synchronized')
+    syncLog.info('Device synchronized')
   } catch (error) {
-    log.warn('Error while synchronizing device', error)
+    syncLog.warn('Error while synchronizing device', error)
+  }
+}
+
+export const getClientName = async (): Promise<string> =>
+  `${SOFTWARE_NAME} (${await getDeviceName()})`
+
+export const checkClientName = async (client: CozyClient): Promise<void> => {
+  const { clientName, ...oauthOptions } = client.getStackClient().oauthOptions
+
+  if (!clientName.startsWith(SOFTWARE_NAME)) {
```

Comment:
Here we can compare `clientName` to the entire `getClientName`, so it will also update if the user changes their device's name

Reply:
indeed, didn't think of that. 

- [ ] updated